include/s3select.h

```diff
@@ -2174,6 +2176,8 @@ class base_s3object
   scratch_area* m_sa;
   std::string m_obj_name;
   bool m_aggr_flow = false; //TODO once per query
+  bool is_star = false;
+  bool is_json_query = false;
```

Comment:
its better to add `class enum` (CSV,Parquet,JSON) to `base_s3select`
upon creating the derived(of `base_s3select`) objects, it sets the type of query.
this way we save redundant methods, and avoid conflicts and bugs.